packages/hmr-plugin/src/internal/hmr-state-context-factory.ts

```diff
@@ -0,0 +1,43 @@
+import { StateContext, Store } from '@ngxs/store';
+import { isStateOperator } from '@ngxs/store/operators';
+import { NgModuleRef } from '@angular/core';
+
+export class HmrStateContextFactory<T, S> {
+  public store: Store;
+
+  constructor(module: NgModuleRef<T>) {
+    const store = module.injector.get(Store, null);
+
+    if (!store) {
+      throw new Error('Store not found, maybe you forgot to import the NgxsModule');
+    }
+
+    this.store = store;
+  }
+
+  /**
+   * @description
+   * must be taken out into  @ngxs/store/internals
```

Comment:
Are you leaving this for a future PR?

Reply:
Could be left until some future refactoring.

Reply:
Any PR